frame/multisig/src/lib.rs

```diff
@@ -539,6 +678,14 @@ impl<T: Config> Pallet<T> {
 			let timepoint = maybe_timepoint.ok_or(Error::<T>::NoTimepoint)?;
 			ensure!(m.when == timepoint, Error::<T>::WrongTimepoint);
 
+			// Ensure that the mutlisig did not expire.
+			if let Some(expiry) = <MultisigExpiries<T>>::get(&id, call_hash) {
+				ensure!(
+					expiry >= <frame_system::Pallet<T>>::block_number(),
+					Error::<T>::MultisigExpired
+				);
+			}
```

Comment:
is there a reason the user wouldn't always want to cleanup storage to get their deposit back here?

Reply:
Not really, the only reason I can think of is to keep the `operate` function more simple since it already has a bunch of logic. 
But if you think  I should add the cleanup here I can create a separate `do_cleanup` function which would be called here and inside `cancel_as_multi`. I actually think this could be reasonable to do.

Reply:
We could actually do something interesting like transferring the user's deposit to the person who cleans up the operation.

Reply:
> Not really, the only reason I can think of is to keep the `operate` function more simple since it already has a bunch of logic. But if you think I should add the cleanup here I can create a separate `do_cleanup` function which would be called here and inside `cancel_as_multi`. I actually think this would be reasonable to do.

This seems like a superior UX to me. Unless there's a reason not to, I think let's cleanup here then I think along with the other change, we can remove `clear_multi` entirely.